sdk/storage/Azure.Storage.Blobs.Batching/src/DelayedResponse.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Azure.Core.Http;
+using Azure.Core.Pipeline;
+
+namespace Azure.Storage.Blobs.Specialized
+{
+    /// <summary>
+    /// A Response that can be returned before a request is actually sent and
+    /// will throw until a live response is provided to wrap.
+    /// </summary>
+    internal class DelayedResponse : Response
```

Comment:
This makes me slightly sad but if that's the pattern we decided on...

Reply:
The implementation?  Or returning "exploding Responses"?  The idea was most customers will use the higher level `BlobBatchClient.DeleteBlobs` and these will only be returned for fine grained error tracking/control from `BlobBatch.DeleteBlob`.  I'm open to any suggestion for improvements that can be made before we ship.

Reply:
`exploding Responses`. It cuts on the number of new types and the feature itself is pretty niche so I think it's fine.

Reply:
> It cuts on the number of new types

That's exactly how we got here.  😄